presentation/components/src/presentation-components/tree/controlled/PresentationTreeNodeRenderer.tsx

```diff
@@ -22,25 +24,44 @@ export function PresentationTreeRenderer(props: TreeRendererProps) {
   />;
 }
 
+interface FilterButtonProps {
+  onFilter?: (node: TreeModelNode) => void;
+}
+
+const FilterButton = (props: FilterButtonProps) => {
+  return (
+    <IconButton className="presentation-filter-action-button" styleType="borderless"
+      onClick={() => props.onFilter}>
+      <SvgFilter className="presentation-filter-icon"/>
+    </IconButton>
+  );
+};
+
 /**
  * @alpha
  */
 export interface PresentationTreeNodeRendererProps extends TreeNodeRendererProps {
-  nodeRenderer?: (props: TreeNodeRendererProps) => React.ReactNode;
+  onFilter?: (node: TreeModelNode) => void;
 }
 
 /**
  * @alpha
  */
 export function PresentationTreeNodeRenderer(props: PresentationTreeNodeRendererProps) {
-  const { nodeRenderer, ...restProps } = props;
+  const { onFilter, ...restProps } = props;
+  const className = classnames("presentation-node", restProps.className);
   if (restProps.node.item.extendedData !== undefined && restProps.node.item.extendedData.tooManyChildren === true)
     return <TooManyChildNodeRenderer
       nodeDepth={restProps.node.depth}
     />;
-  return nodeRenderer ? <>{nodeRenderer(restProps)}</> : <TreeNodeRenderer
-    {...restProps}
-  />;
+  return <TreeNodeRenderer
+    {...restProps }
+    className={className}
+  >
+    <FilterButton
```

Comment:
It looks like there is no need for separate `FilterButton` component as all it does just renders button